Makefile.dep

```diff
@@ -629,6 +629,20 @@ ifneq (,$(filter spiffs,$(USEMODULE)))
     USEMODULE += mtd
 endif
 
+ifneq (,$(filter fatfs_vfs,$(USEMODULE)))
+    USEPKG += fatfs
+    USEMODULE += vfs
+
+    ifeq ($(BOARD),native)
+      USEMODULE += fatfs_diskio_native
+      FATFS_DISKIO_NATIVE_DEFAULT_FILE ?= \"riot_fatfs_disk.img\"
+    else
+      USEMODULE += fatfs_diskio_sdcard_spi
+      USEMODULE += auto_init_storage
+    endif
```

Comment:
I disagree on this one.
1: "that means a lot of duplication" and "seems to be cleaner" sounds somewhat contrary to me. 
2: the fact that FatFs (for now) can only be used with either an image file on native or with sdcard_spi is a restriction that relates directly to the FatFs diskio layer and not to a specific board.
3: the restriction of 2 is likely to be removed in the future by adding a more dynamic diskio layer (e.g. think of using FAT formatted USB sticks and sdcards simultaneously). This would then lead to removing this duplicated code from all boards makefiles again.
4: IMO a dev who adds a new board (and its Makefile.dep) shouldn't have to think about "weird FatFS-internal" stuff.
But I should indeed move the board dependent ifdef to the fatfs Makefile.dep instead.